actions/docs-verifier/src/ActionRunner/Program.cs

```diff
@@ -56,15 +70,14 @@ static bool IsRedirectableFile(PullRequestFile file)
 
     // A deleted toc.yml doesn't need redirection.
     // Also, don't require a redirection for file patterns specified as "exclude"s in docfx config file.
-    return !isDeletedToc && IsYmlOrMarkdownFile(deletedFileName) && !IsInWhatsNewDirectory(deletedFileName) &&
-        DocfxConfigurationReader.GetMatchers().Any(m => m.Match(deletedFileName).HasMatches);
+    return !isDeletedToc && IsYmlOrMarkdownFile(deletedFileName) && !IsInWhatsNewDirectory(deletedFileName, whatsNewPath) &&
+        matchers.Any(m => m.Match(deletedFileName).HasMatches);
 }
 
 static bool IsYmlOrMarkdownFile(string? fileName) => Path.GetExtension(fileName) is ".yml" or ".md";
 
-static bool IsInWhatsNewDirectory(string? fileName)
+static bool IsInWhatsNewDirectory(string? fileName, string? whatsNewPath)
```

Comment:
Minor suggestion: I think `fileName` can be made non-nullable, then we won't need null conditional access nor `== true`. For that to be done, I think we'd need to add `[NotNullWhen(true)]` to `fileName` in `IsYmlOrMarkdownFile`.

Reply:
Awesome, you're correct - good catch!